src/zmi/styles/subscriber.py

```diff
@@ -1,23 +1,61 @@
+import itertools
+
 import zope.component
 import zope.interface
+from Acquisition import aq_parent
+from ZPublisher.HTTPRequest import HTTPRequest
+
+
+def prepend_authentication_path(request: HTTPRequest, path: str) -> str:
+    """Prepend the path of the user folder.
+
+    Because ++resource++zmi is protected, we generate URLs relative to the
+    user folder of the logged-in user, so that the user can access the
+    resources.
+    """
+    authenticated_user = request.get("AUTHENTICATED_USER")
```

Comment:
Please do not pick the user from the `REQUEST`. That's convenient but a request can be manipulated. Do it like this instead:

```
from AccessControl.SecurityManagement import getSecurityManager
<...>
authenticated_user = getSecurityManager().getUser()
```

Reply:
Thanks ! I amended the commit accordingly